### DISCUSSION TITLE
`No Cargo projects found` is confusing

### DISCUSSION THREAD

**Discussion by vjpr** [intellij-rust/intellij-rust#7808] [Other]
When I start a new project from the terminal, when I open my `main.rs` I get this message.

Why not just pick the first `Cargo.toml` file it finds? Or prompt me if this is the one I want to add?

Is it because a project may contain many Cargo packages?

For a new user, when you click `Attach` you are unsure what you should select. I only found out from a comment on a GH issue. And the starting directory of the `Attach` dialog is the user's home, which makes it painful to navigate to the correct dir.

Also, after clicking attach, where is this information stored? I was expecting it to show up as a `Module` in the `Module Settings`.

**Comment by vlad20012** [comment]
> Also, after clicking attach, where is this information stored? I was expecting it to show up as a `Module` in the `Module Settings`

In the [cargo tool window](https://plugins.jetbrains.com/plugin/8182-rust/custom-pages/docs/rust-cargo-tool-window.html)

![image](https://user-images.githubusercontent.com/3221931/132379407-6a9cd725-2285-4a59-af3f-56b0fa8250b6.png)

**Comment by merriam** [comment]
So a better message might be "No Cargo file found. Select a Cargo.toml file".
A better starting directory would be the project's base.
An even better solution would be to automatically select 'Cargo.toml' if it is the only toml file in the project's base directory.
Another even better additio would be, if the project base directory ends in "src" and the immediate parent has a Cargo.toml, then append to the message "Did you mean to create the project in the parent directory?"

**Comment by merriam** [comment]
Maybe it means something totally different.  I just crashed PyCharm.

Using Rust plugin.  Got odd ‘No Cargo projects found’ message (bug #7808).  Attached the project directory.  Still had the message.  Tried attaching the project directory’s Cargo.toml file.  Still had the message.  Tried to attach it again.  Hung with beachball of death.

**Reply by felipe-bravox** [reply]
Same here

**Comment by ghenry** [comment]
Confused me too creating a new project in CLion.

**Comment by zkxjzmswkwl** [comment]
This should just be a thing.

**Comment by kkshinkai** [comment]
This also confused me.

And after I select the `Config.toml` file, the problem is not solved. I tried `Attach` many times, it didn't work for me.

I restarted IDEA and dragged a new project in. IDEA started indexing the Rust library (You can see the progress bar of the task at the bottom), and then everything worked fine. It's probably some weird cache or config issue.

**Reply by XmchxUp** [reply]
> This also confused me.
> 
> And after I select the `Config.toml` file, the problem is not solved. I tried `Attach` many times, it didn't work for me.
> 
> I restarted IDEA and dragged a new project in. IDEA started indexing the Rust library (You can see the progress bar of the task at the bottom), and then everything worked fine. It's probably some weird cache or config issue.

it works.

**Comment by seeekr** [comment]
Invalidating caches and restarting worked for me.

**Reply by Nirvanalist** [reply]
This is work for me.
I create  a file that contained others registry named config in .cargo dir, restart the IDE and wait a minute, it works

**Comment by turtletongue** [comment]
[Attaching cargo.toml with context action](https://plugins.jetbrains.com/plugin/8182-rust/docs/attaching-cargo-projects.html) works for me

**Comment by pythonaddict** [comment]
You should wait for a few minutes, the IDE is configuring your project. When this process ends, that message disappears by itself.

**Reply by johnperry-math** [reply]
I believe it, but this is both terrible behavior and a misleading message. It needs to change.
